cpp/libcugraph_etl/src/renumbering.cu

```diff
@@ -16,9 +16,954 @@
+
+struct renumber_functor {
+  template <typename Dtype,
+                std::enable_if_t<not std::is_integral<Dtype>::value>* = nullptr>
+  std::tuple<std::unique_ptr<cudf::column>,
+              std::unique_ptr<cudf::column>,
+              std::unique_ptr<cudf::table>> operator() (cudf::table_view const& src_view,
+                                            cudf::table_view const& dst_view) {
+    return std::make_tuple(std::unique_ptr<cudf::column>(new cudf::column(
+                                                            cudf::data_type(cudf::type_id::INT32),
+                                                            0,
+                                                            rmm::device_buffer{0,cudaStream_t{0}})
+                                                          ),
+                           std::unique_ptr<cudf::column>(new cudf::column(
+                                                            cudf::data_type(cudf::type_id::INT32),
+                                                            0,
+                                                            rmm::device_buffer{0,cudaStream_t{0}})
+                                                          ),
+                            std::make_unique<cudf::table>(std::vector<std::unique_ptr<cudf::column>>{}));
+  }
+
+  template <typename Dtype,
+                std::enable_if_t<std::is_integral<Dtype>::value>* = nullptr>
+  std::tuple<std::unique_ptr<cudf::column>,
+        std::unique_ptr<cudf::column>,
+        std::unique_ptr<cudf::table>> operator() (cudf::table_view const& src_view,
+                                            cudf::table_view const& dst_view) {
+    assert(src_view.num_columns() == 2);
+    assert(dst_view.num_columns() == 2);
+
+    size_type num_rows = src_view.num_rows();
+    using char_type = int8_t;
+    using str_offset_type = int32_t;  // kernels init'd int32 only
+
+    std::vector<char_type*> src_vertex_chars_ptrs;
+    std::vector<str_offset_type*> src_vertex_offset_ptrs;
+    std::vector<char_type*> dst_vertex_chars_ptrs;
+    std::vector<str_offset_type*> dst_vertex_offset_ptrs;
+
+    for (int i = 0; i < src_view.num_columns(); i++) {
+      auto str_col_view = cudf::strings_column_view(src_view.column(i));
+      src_vertex_chars_ptrs.push_back(const_cast<char_type*>(str_col_view.chars().data<char_type>()));
+      src_vertex_offset_ptrs.push_back(const_cast<str_offset_type*>(
+                                            str_col_view.offsets().data<str_offset_type>()));
+    }
+
+    for (int i = 0; i < dst_view.num_columns(); i++) {
+      auto str_col_view = cudf::strings_column_view(dst_view.column(i));
+      dst_vertex_chars_ptrs.push_back(const_cast<char_type*>(str_col_view.chars().data<char_type>()));
+      dst_vertex_offset_ptrs.push_back(const_cast<str_offset_type*>(
+                                            str_col_view.offsets().data<str_offset_type>()));
+    }
+
+    accum_type *hist_insert_counter = nullptr;
```

Comment:
Can we make this an std::vector?  The RAAI construct is generally cleaner and easier to maintain.

The cugraph code uses raft utilities to support easy transfers between host and device.  https://github.com/rapidsai/raft/blob/6a8c7a3bebe85d8fef34951e6f09c93fa733b06f/cpp/include/raft/cudart_utils.h#L251

Reply:
In order to efficiently use raft utilities, we *might* need to add the `raft::handle_t` as a parameter to the renumber_cudf_tables method.

Reply:
Changing this to a raft::mr::host::buffer for RAII. I directly write some values to pinned memory so can't use a std::vector.